### PR TITLE
Remove fuse

### PR DESCRIPTION
**Pull request by ebassity**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

Fuse.js was overkill for `<Select>` filtering. Pulled it out and replaced with the case-insensitive substring approach used by most component libraries. Also fixes an `aria-required-children` 508 compliance violation in TabList.

**Changes**
  - removed deps: `fuse.js`, `@reaviz/react-use-fuzzy`
  - Select.tsx: `useFuzzy` block replaced with `useState` + memoized `filterOptionsByKeyword` (substring match on `children` and `group`, case-insensitive).
  - `SelectProps.searchOptions` removed
  - Filter extracted to `utils/filter.ts` matching the existing `utils/{grouping,options,keyboard}.ts` pattern; spec imports the real function (no test/production drift risk).
  - MultiSelect Createable story: dropped `searchOptions={{ threshold: 0 }}`
  - TabList.tsx: decorative `<hr>` divider now has `aria-hidden="true"`, resolving axe `aria-required-children` violation on `<nav role="tablist">`

**Version**
Minor bump: 10.0.3 -> 11.0.0

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[x] Other... Please describe: dependency removal + behavior simplification with public-API removal
```

## What is the current behavior?

Fuse.js was overkill for `<Select>` filtering. Pulled it out and replaced with the case-insensitive substring approach used by most component libraries. Also fixes an `aria-required-children` 508 compliance violation in TabList.

## What is the new behavior?
  - No more typo tolerance: `"Aplee"` no longer matches `"Apple"`.
  - No more multi-word tokenization: `"App Fr"` no longer matches grouped
    `"Apple / Fruit"` entries.

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

## Other information
**Verification**
  - 171 unit tests pass (165 existing + 6 new substring tests covering the contract, ordering, and missing/non-string-children edge cases)
  - `npm run build:js` clean
  - `npm ls fuse.js @reaviz/react-use-fuzzy` → both empty
  - reablocks dist unchanged in size (Fuse was already external); savings land in consumer bundles via removed transitive deps

**Follow-ups**
  - Future: consider `filterFn` extension point if/when a consumer needs custom matching (i18n normalization, ReactNode labels, alternative fields, etc.).

### CODE REVIEW

File: src/form/Select/Select.tsx [hunk 1]
```diff
@@ -245,9 +249,12 @@ export interface SelectProps {
...
 
   /**
-   * The options for the Fuse.js search algorithm.
+   * @deprecated The Fuse.js-based fuzzy search has been replaced with
+   * case-insensitive substring matching on the option label and group.
+   * This prop is now a no-op and will be removed in the next major
+   * release. Remove it from your code.
    */
-  searchOptions?: Fuse.IFuseOptions<SelectOptionProps>;
+  searchOptions?: Record<string, unknown>;
```
Comment (amcdnl) [inline src/form/Select/Select.tsx]: Lets just remove it - we don't want users to be fooled by this is doing something but not really.

File: src/form/Select/Select.tsx [hunk 2]
```diff
@@ -328,19 +334,16 @@ export const Select: FC<SelectProps> = ({
...
-    resetSearch
-  } = useFuzzy<SelectOptionProps>(options, {
-    keys: ['children', 'group'],
-    ...searchOptions,
-    getFn: menuDisabled ? () => '' : searchOptions?.getFn
-  });
-
-  // TODO: Come back and cleanup the fuzzy search to be more extensible
-  const result = filterable === 'async' ? options : fuseResult;
+  const [keyword, setKeyword] = useState('');
```
Comment (amcdnl) [inline src/form/Select/Select.tsx:329]: I was thinking about making a generic hook that had a similar interface but didn't call fuse. I think this MIGHT also be needed by the command palette too.

**Review by amcdnl — CHANGES_REQUESTED** [review]
Minor tweaks/feedback

File: src/form/Select/Select.tsx (hunk 2 above)
Comment (ebassity) [reply src/form/Select/Select.tsx:329]: @amcdnl - CommandPalette has a search input that emits onSearchChange but doesn't filter internally; filtering, if needed, lives in the parent(the onSearchChange callback + parent-decides-filtering pattern). adopting an internal filter hook would mean either changing its API to take items data instead of children (breaking) or adding optional internal filtering alongside the existing pattern (more surface, ambiguous).

unless you feel strongly, since Select is the only confirmed consumer right now, i might old off on the abstraction. lmk!

File: src/form/Select/Select.tsx (hunk 1 above)
Comment (ebassity) [reply src/form/Select/Select.tsx]: removed!

File: src/form/Select/Select.spec.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      { value: 'b', children: null as unknown as string, group: 'fruit' },
+      { value: 'c', children: 'banana', group: 'fruit' }
+    ];
+    expect(
+      filterOptionsByKeyword(edgeOptions, 'banana').map(o => o.value)
+    ).toEqual(['c']);
+    expect(
+      filterOptionsByKeyword(edgeOptions, 'fruit').map(o => o.value)
+    ).toEqual(['a', 'b', 'c']);
+  });
```
[the remark this answers is not in the document]
Comment (ebassity) [reply src/form/Select/Select.spec.ts:57]: resolved in https://github.com/reaviz/reablocks/pull/357/commits/7db84c11a6e16d53d02b6a349f7947a8cbc01f5a

File: src/form/Select/utils/filter.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  keyword: string
+): SelectOptionProps[] => {
+  if (!keyword) return options;
+  const q = keyword.toLowerCase();
+  return options.filter(
+    o =>
+      String(o.children ?? '')
+        .toLowerCase()
+        .includes(q) || (o.group ?? '').toLowerCase().includes(q)
+  );
```
[the remark this answers is not in the document]
Comment (ebassity) [reply src/form/Select/utils/filter.ts]: resolved in https://github.com/reaviz/reablocks/pull/357/commits/7db84c11a6e16d53d02b6a349f7947a8cbc01f5a
